static/app/views/issueList/editableTabTitle.tsx

```diff
@@ -0,0 +1,87 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+function EditableTabTitle({
+  label,
+  onChange,
+  isEditing,
+  setIsEditing,
+}: {
+  isEditing: boolean;
+  label: string;
+  onChange: (newLabel: string) => void;
+  setIsEditing: (isEditing: boolean) => void;
+}) {
+  const [inputValue, setInputValue] = useState(label);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const isEmpty = !inputValue.trim();
+
+  const handleOnBlur = () => {
+    if (!isEditing) {
+      return;
+    }
+    if (isEmpty) {
+      setInputValue(label);
```

Comment:
You could avoid this line if the input (and corresponding state) was extracted to its own component. That way the state would be reset whenever the edit state was entered.